oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java

```diff
@@ -365,7 +370,14 @@ static void decode(final UserAgent target, final String[] args, final InputStrea
     }
 
     static String extractResponseFromRedirectUri(final String redirectedUri) {
-        final URI uri = URI.create(redirectedUri);
-        return uri.getQuery();
+        final URL uri;
+        try {
+            uri = new URL(null, redirectedUri);

```

Comment:
FWIW, a URL constructed this way has the correct query param; URI.create() would generate a null query param.